src/proxy/socks5.rs

```diff
@@ -131,26 +130,56 @@ impl Socks5 {
         .await
     }
 }
-
+async fn handle_socks_connection(mut oc: OutboundConnection, mut stream: TcpStream) {
+    match negotiate_socks_connection(&oc.pi, &mut stream).await {
+        Ok(target) => {
+            // In many cases, when we are fail we don't have this.
+            // TODO: ideally, we send the success after we connect. This allows us to actually give a
+            // success only when we really succeeded, rather than if we completed the SOCKS handshake.
+            // Additionally, it would allow us to get a proper address to send back/
+            let dummy_addr = SocketAddr::new(Ipv4Addr::new(0, 0, 0, 0).into(), 0);
+            if let Err(err) = send_success(&mut stream, dummy_addr).await {
+                warn!("failed to send socks success response: {err}");
+                return;
+            }
+            let remote_addr =
+                socket::to_canonical(stream.peer_addr().expect("must receive peer addr"));
+            oc.proxy_to(stream, remote_addr, target).await
+        }
+        Err(e) => {
+            warn!("failed to negotiate socks connection: {e}");
+            send_error(&e, &mut stream).await;
+        }
+    }
+}
 // handle will process a SOCKS5 connection. This supports a minimal subset of the protocol,
```

Comment:
This comment needs updating, took a crack at it but it can probably still be better worded
```suggestion
// negotiate_socks_connection will inspect a SOCKS5 connection to produce a socket address for supported connections. This supports a minimal subset of the protocol,
```